src/frontend/src/utils/condition.rs

```diff
@@ -849,6 +850,7 @@ impl Condition {
             .into_iter()
             .map(push_down_not)
             .map(fold_boolean_constant)
+            .map(simplify_stream_filter_expression)
```

Comment:
If we put `simplify_filter_expression ` in the Condition simplify function, we need to remember that never treat null as false, because users could use this condition (A) to `and` another condition(B), so the result of `A and B` should follow that sheet. 

| A & B   | A=true | A=false | A=null |
| ------- | ------ | ------- | ------ |
| B=true  | true   | false   | null   |
| B=false | false  | false   | false  |
| B=null  | null   | false   | null   |

Reply:
However, our target is to simplify the filter expression, so we can implement an optimization rule to ensure that the expression is the filter condition. And then we can treat the final null as a false.

Reply:
Let's wait for #15373 to be merged, and presumably `Strong` is the all-in-one solution to this expression rule.
i.e., only call `Strong::is_null(expr)` should be enough for any potential `null` cases' checking.

Reply:
> condition (A) to and another condition(B)

Do we have a concrete example for this scenario? Especially not in filter predicate.